docs/concepts/glossary.md

```diff
@@ -0,0 +1,140 @@
+[Operator](./glossary#operator).
+
+#### Instance
+
+The term *instance* is used to describe a specific instance of a specific type (usually
+[Operator](./glossary#operator) or [Function](./glossary#function)) during runtime. As Apache Flink
+is mostly written in Java, this corresponds to the definition of *Instance* or *Object* in Java.
+In the context of Apache Flink, the term *parallel instance* is also frequently used to emphasizes
+that multiple instances of the same [Operator](./glossary#operator) or
+[Function](./glossary#function) type are running in parallel.
+
+#### Flink Job
+
+A Flink Job is the runtime representation of a Flink program. A Flink Job can be either submitted
+to a long running [Flink Session Cluster](./glossary#flink-session-cluster) or it can be started as a
+self-contained [Flink Application Cluster](./glossary#flink-application-cluster).
+
+#### JobGraph
+
+see [Logical Graph](./glossary#logical-graph)
+
+#### Logical Graph
+
+A logical graph is a directed graph describing the high-level logic of a stream processing program.
+The nodes are [Operators](./glossary#operator) and the edges indicate input/output-relationships or
+data streams or data sets.
+
+#### Managed State
+
+Managed State describes application state, which has been registered with the framework. For
+Managed State Apache Flink will take care about persistence and rescaling among other things.
+
+#### Operator
+
+Node of a [Logical Graph](./glossary#logical-graph). An Operator performs a certain operation,
+which is usually executed by a [Function](./glossary#function). Sources and Sinks are special
+Operators for data ingestion and data egress.
+
+#### Partition
+
+A partition is an independent subset of the overall data stream or data set. A data stream or
+data set is divided into partitions by assigning each [record](./glossary#Record) to one or more
+partitions. Partitions of data streams or data sets are consumed by [Tasks](./glossary#task) during
+runtime.
+
+#### Physical Graph
+
+A physical graph is the result of translating a [Logical Graph](./glossary#logical-graph) for
+execution in a distributed runtime. The nodes are [Tasks](./glossary#task) and the edges indicate
+input/output-relationships or partitions of data streams or data sets.
+
+#### Record
+
+Records are the constituent elements of a data set or data stream.
+[Operators](./glossary#operator) and [Functions](./glossary#Function) receive records as input
+and emit records as output.
+
+#### Flink Session Cluster
+
+A long-running [Flink Cluster](./glossary#flink-cluster), which accepts multiple
+[Flink Jobs](./glossary#flink-job) for execution. The lifetime of this Flink Cluster is not bound
+to the lifetime of any Flink Job. Formerly, a Flink Session Cluster was also known as a Flink Cluster in
+*session mode*.
+
+#### State Backend
+
+For stream processing programs the State Backend determines how state is stored on each Taskmanager
```

Comment:
"For stream processing programs,..." (with comma)

Reply:
Done